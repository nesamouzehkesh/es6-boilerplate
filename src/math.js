export function add (a, b) {
  return a + b;
}

export function mul (a, b) {
  return a * b;
}

export function complex (n) {
  return [...Array (n).keys ()].map (index => ({
    id: index,
    name: `user ${index}`,
  }));
}
