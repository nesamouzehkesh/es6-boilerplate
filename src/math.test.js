import {add, complex} from './math';

describe ('Testing math service', () => {
  test ('test add function', () => {
    expect (add (2, 5)).toEqual (7);
    expect (add (2, 5)).toMatchSnapshot ();
  });

  test ('test complex function', () => {
    expect (complex (10)).toMatchSnapshot ();
  });
});
