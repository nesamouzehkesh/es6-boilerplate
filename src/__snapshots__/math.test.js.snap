// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing math service test add function 1`] = `7`;

exports[`Testing math service test complex function 1`] = `
Array [
  Object {
    "id": 0,
    "name": "user 0",
  },
  Object {
    "id": 1,
    "name": "user 1",
  },
  Object {
    "id": 2,
    "name": "user 2",
  },
  Object {
    "id": 3,
    "name": "user 3",
  },
  Object {
    "id": 4,
    "name": "user 4",
  },
  Object {
    "id": 5,
    "name": "user 5",
  },
  Object {
    "id": 6,
    "name": "user 6",
  },
  Object {
    "id": 7,
    "name": "user 7",
  },
  Object {
    "id": 8,
    "name": "user 8",
  },
  Object {
    "id": 9,
    "name": "user 9",
  },
]
`;
